src/main/java/net/minestom/server/extras/bungee/BungeeCordProxy.java

```diff
@@ -24,4 +34,45 @@ public static void enable() {
     public static boolean isEnabled() {
         return enabled;
     }
+
+    /**
+     * Sets the tokens used by BungeeGuard authentication.
+     * Setting the tokens to a not-null value enables BungeeGuard authentication,
+     * and setting it to a null value disables BungeeGuard authentication.
+     *
+     * @param tokens The new BungeeGuard authentication tokens
+     */
+    public static void setBungeeGuardTokens(@Nullable Set<String> tokens) {
+        bungeeGuardTokens = tokens;
+    }
+
+    /**
+     * Checks whether BungeeGuard authentication is enabled.
+     *
+     * @return Whether BungeeGuard authentication is enabled
+     */
+    public static boolean isBungeeGuardEnabled() {
+        return bungeeGuardTokens != null;
+    }
+
+    /**
+     * Checks whether a token is one of the valid BungeeGuard tokens
+     *
+     * @param token The token to test
+     * @return Whether the token is a valid BungeeGuard token
+     */
+    public static boolean isValidBungeeGuardToken(@NotNull String token) {
+        return isBungeeGuardEnabled() && bungeeGuardTokens.contains(token);
+    }
+
+    /**
+     * Gets the maximum length of the handshake packet.
+     *
+     * @return The maximum length of the handshake packet
+     */
+    public static int getMaxHandshakeLength() {
```

Comment:
Looks like an implementation detail of the handshake packet, and no need to be exposed